addons/website/static/src/snippets/s_facebook_page/000.js

```diff
@@ -24,36 +24,53 @@ const FacebookPageWidget = publicWidget.Widget.extend({
             params.href = `https://www.facebook.com/${params.id}`;
         }
         delete params.id;
-        params.width = utils.confine(Math.floor(this.$el.width()), 180, 500);
 
-        var src = $.param.querystring('https://www.facebook.com/plugins/page.php', params);
-        this.$iframe = $('<iframe/>', {
-            src: src,
-            width: params.width,
-            height: params.height,
-            css: {
-                border: 'none',
-                overflow: 'hidden',
-            },
-            scrolling: 'no',
-            frameborder: '0',
-            allowTransparency: 'true',
-        });
-        this.$el.append(this.$iframe);
+        this._renderIframe(params);
+        this.resizeObserver = new ResizeObserver(debounce(this._renderIframe.bind(this, params), 100));
+        this.resizeObserver.observe(this.el.parentElement);
 
-        this.options.wysiwyg && this.options.wysiwyg.odooEditor.observerActive();
         return def;
     },
     /**
      * @override
      */
     destroy: function () {
         this._super.apply(this, arguments);
+        this.resizeObserver.disconnect();
+    },
 
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Prepare iframe element & replace it with existing iframe.
+     *
+     * @private
+     * @param {Object} params
+    */
+    _renderIframe: function(params) {
```

Comment:
```suggestion
    _renderIframe(params) {
```
@paru-odoo Could you change, thanks :pray: